### ISSUE TITLE
Sending to untrusted identity contact: exit code

### ISSUE THREAD

**Issue by exquo** [AsamK/signal-cli#828]
Sending a message to an "untrusted identity" contact (e.g. one who reinstalled signal), signal-cli returns a `0` ("success") exit code:  

	signal-cli send -m 'message text' +1234567890; echo $?

	Failed to send (some) messages:
	+1234567890: Untrusted Identity for "+1234567890"
	1630000000000
	0

I assume the exit code should be non-zero (failure), since the message is not actually sent.

When sending a message through a dbus API, it does return non-zero code (`2`): 

	signal-cli -u +0987654321 --output=json daemon &
	signal-cli --dbus send -m 'msg text' +1234567890; echo $?

	Untrusted identity: Untrusted identity: 1630000000000
	Failed to send message:
	Untrusted Identity for "+1234567890"
	2

Although, in the man page, ["Exit codes" section](https://github.com/AsamK/signal-cli/blob/d1ccd11fcb7d5657f0b2a96af0941c1e4af00a6d/man/signal-cli.1.adoc#exit-codes) it says

> *4*: Sending failed due to untrusted key

When sending to a *group*, one of whose members has reinstalled signal, it makes sense if the return is "success", as per #348 and https://github.com/AsamK/signal-cli/commit/06e93b84da2718c31111e820cd35a3354f22bae2. This is indeed what happens with non-dbus execution:

	signal-cli send -g 'txs4no+ebx<...GROUP_ID..>iVebi74gv+riVoBi7o=' -m 'non-dbus send to group'; echo $? 

	Failed to send (some) messages:
	txs4no+ebx<...GROUP_ID..>iVebi74gv+riVoBi7o=: Untrusted Identity for "+1234567890"
	1630000000000
	0

But running the group send through dbus results in non-zero exit code:

	signal-cli -u +0987654321 --output=json daemon &
	signal-cli --dbus send -g 'txs4no+ebx<...GROUP_ID..>iVebi74gv+riVoBi7o=' -m 'dbus send to group'; echo $?

	Failure: Failure: 1630000000000
	Failed to send (some) messages:
	txs4no+ebx<...GROUP_ID..>iVebi74gv+riVoBi7o=: Untrusted Identity for "+1234567890"
	2

To sum up:

- Expected result: 
	- sending to untrusted *individual* (either through dbus or CLI) -> exit code non-zero
	- sending to a *group* with untrusted individual (either through dbus or CLI) -> exit code zero.
- Actual result: 
	- sending through CLI (to either individual or group) -> exit code zero
	- sending through dbus (to either individual or group) -> exit code non-zero.

**Comment by exquo** [comment]
Thanks for implementing this! I have tried the latest `master` and the exit codes work as expected.

A small (and I assume unintentional) wrinkle is that now sending to a group through dbus does not give any indication that one of the members is unregistered:

	signal-cli -u +0987654321 --output=json daemon &
	signal-cli --dbus send -g 'txs4no+ebx<...GROUP_ID..>iVebi74gv+riVoBi7o=' -m 'dbus send to group'; echo $?

	1640000000000
	0

Non-dbus `send` to group *does* print out a helpful notification, as before:

	signal-cli send -g 'txs4no+ebx<...GROUP_ID..>iVebi74gv+riVoBi7o=' -m 'non-dbus send to group'; echo $? 

	Failed to send (some) messages:
	txs4no+ebx<...GROUP_ID..>iVebi74gv+riVoBi7o=: Untrusted Identity for "+1234567890"
	1640000000000
	0

I think it would make sense to restore the `Failed …` output that has been generated for the 'dbus' send case before, analogous to what is printed out for non-dbus command.

**Comment by AsamK** [comment]
That's a limitation of the current dbus interface. The send method there can either return a message timestamp or and error message. It would need to be extended to return additional information (and should then also support sending quotes and mentions ...)

**Comment by exquo** [comment]
I see, thanks!

Support for sending mentions and quotes through dbus would be very useful!

Incidentally, I would have guessed that sending messages containing quotes or mentions would be no different from messages without those additional flags to the `send` command. But I suppose there might be situations where some output is printed, yet those are not errors.
